### PR TITLE
Phase2-hgx122 Add DetId for a new proposed detector (HFNose)

### PR DESCRIPTION
**Pull request by bsunanda**
This new detector proposed to be placed in front of HF during Run4 operation need evaluation for Jet/MET studies. Also cleaned up some of the legacy non-functional code.

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by kpedro88** [comment]
@bsunanda a few questions:
1. who is proposing this? is there some explanation or justification?
2. was `HGCHET` ever assigned to any DetIds that could have been stored in officially produced files?

**Comment by bsunanda** [comment]
HGCHET was never used and HFNose is a detector proposed for Phase2 upgrade and needs studies to justify its inclusion

**Comment by kpedro88** [comment]
+1

File: DataFormats/ForwardDetId/interface/ForwardSubdetector.h [hunk 1]
```diff
@@ -2,7 +2,7 @@
 #define DataFormats_ForwardDetId_ForwardSubDetector_H 1
 
 enum ForwardSubdetector { ForwardEmpty=0, FastTime=1, BHM=2, HGCEE=3, HGCHEF=4,
-			  HGCHEB=5, HGCHET=6, HGCTrigger=7 };
+			  HGCHEB=5, HFNose=6, HGCTrigger=7 };
```
[the remark this answers is not in the document]
Comment (kpedro88) [reply DataFormats/ForwardDetId/interface/ForwardSubdetector.h]: I'm not sure this PR is the place to change the type of `ForwardSubdetector` used throughout CMSSW...

File: DataFormats/ForwardDetId/interface/HFNoseDetId.h [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+                 1 coarse divisions of wafer with 200 mum thick silicon
+                 2 coarse divisions of wafer with 300 mum thick silicon)
+   [25:27] Subdetector type (HFNose)
+   [28:31] Detector type (Forward)
+*/
+class HFNoseDetId : public DetId {
+
+public:
+
+  enum waferType {HFNoseFine=0, HFNoseCoarseThin=1, HFNoseCoarseThick=2};
```
[the remark this answers is not in the document]
Comment (bsunanda) [reply DataFormats/ForwardDetId/interface/HFNoseDetId.h]: It is coded as integers in DetId - better remain as enum's

**Comment by kpedro88** [comment]
+1

**Comment by ianna** [comment]
+1

**Comment by civanch** [comment]
+1

**Comment by fabiocos** [comment]
+1

it affects only Phase2, the removed DetId was not used in code

**Pull request merged**